sdk/src/main/java/velocitas/sdk/AsyncResult.kt

```diff
@@ -0,0 +1,118 @@
+package velocitas.sdk
+
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * Empty result structure which can be used in case an AsyncResult
+ *        does not return any valuable information other than its success.
+ */
+class VoidResult
+
+class AsyncResult<TResultType> {
+    private var result: TResultType? = null
```

Comment:
Yes it needs to be optional, you can't influence the result from the Server. Server obviously could just send out an error message instead of an result